### DISCUSSION TITLE
Sitemap module and custom link

### DISCUSSION THREAD

**Discussion by PiemP** [OrchardCMS/OrchardCore#7689] [Q&A] [answered]
Hi all, 

is it possible to add a custom URL to one sitemap generated by the Sitemap module?
Thank you.

**Comment by deanmarcussen** [comment, marked as answer]
You could probably implement a `ISitemapSourceBuilder` - it has one method on it `BuildAsync` which can be used to add as many elements to a sitemap, as you want.

By elements, I mean xml elements.

Or if it just one url, then it maybe easier, just to pop it in a sitemap.xml file in the wwwroot folder, and refer to it via a sitemapindex.

Equally if you have a lot you want to add, you could implement a `CustomUrlSitemapSource` where you could add urls through the ui.

That would probably be a useful enhancement to the sitemaps module.

**Reply by PiemP** [reply]
clear, thank you @deanmarcussen.

I try to create `CustomUrlSitemapSource` to see if I can do this enhancement: I hope to create soon a new PR about it 😁. At worst I can create a class that implement `ISitemapSourcebuilder`.
